codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/GoIntegration.java

```diff
@@ -111,6 +113,46 @@ default void onShapeWriterUse(
         // pass
     }
 
+    /**
+     * Populates operation middleware stack.
+     *
+     * This method is used to populate middleware stack inside
+     * the operation function.
+     *
+     * @param settings Settings used to generate.
+     * @param model Model to generate from.
+     * @param symbolProvider Symbol provider used for codegen.
+     * @param writer Writer that will be used.
+     * @param serviceShape Service shape of the operation.
+     * @param operationShape Operation Shape for which middleware stack is populated.
+     * @param protocolGenerator Protocol Generator associated with protocol used by the client.
+     */
+    default void assembleMiddlewareStack(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoWriter writer,
+            ServiceShape serviceShape,
+            OperationShape operationShape,
+            ProtocolGenerator protocolGenerator
+    ) {
+        // pass
+    }
+
+    /**
+     * Generates a value for idempotency token.
+     *
+     * @param writer Writer that will be used.
+     * @param operand Operand to which retrieved value should be assigned
+     */
+    default void generateValueForIdempotencyToken(
```

Comment:
Why is idempotency token a special case and not just a normal plugin?

Reply:
Didn't mean to make it a special case. Here we are looking to make the idempotency token value loosely coupled from the generated middleware. I didn't find a way to do this directly without introducing a method. But try to see if this can be a normal plugin.